### DISCUSSION TITLE
Wrapper feature, like mvnw or gradlew

### DISCUSSION THREAD

**Discussion by torrescd** [npm/feedback#551] [CLI] [answered]
Having defined as code the build-tool is not only usefull for CI's or automatic scenarios. 

Be able to perform a checkout and not having to worry about if my global build-tool is compatible or even if I have it is agile.

Some use cases to solve:
 - Having to maintain several projects with different npm cli versions
 - Use exactly the same version of the CI and the rest of the people that works with a given project
 - Having all the needed to build a project in the same project

Sample:
 - https://raw.githubusercontent.com/oliviernt/npmw/master/npmw

Regards

**Comment by arcanis** [comment, marked as answer]
This is what [Corepack](https://github.com/nodejs/corepack) intends to solve. Npm doesn't support it yet, though.
